### PR TITLE
Adds docs for `Array#replace`

### PR DESCRIPTION
**Pull request by wontruefree**
Adds initial docs for Array#replace.

### CODE REVIEW

File: src/array.cr [hunk 1]
```diff
@@ -1381,6 +1381,17 @@ class Array(T)
     self
   end
 
+  # Replaces the contents of the current Array with the contents of the given Array.
```
Comment (straight-shoota) [inline src/array.cr:1384]:
```suggestion
  # Replaces the contents of `self` with the contents of *other*.
```

File: src/array.cr [hunk 2]
```diff
@@ -1381,6 +1381,17 @@ class Array(T)
...
+  # This resizes the Array to a greater capacity but does not free memory if the given array is smaller.
+  # ```
```
Comment (Sija) [inline src/array.cr:1386]:
```suggestion
  #
  # ```
```

**Review by straight-shoota — APPROVED** [review]

File: src/array.cr [hunk 3]
```diff
@@ -1381,6 +1381,18 @@ class Array(T)
...
+  # Replaces the contents of `self` with the contents of *other*.
+  # This resizes the Array to a greater capacity but does not free memory if the given array is smaller.
+  #
+  # ```
+  # a1 = [1, 2, 3]
+  # a1.replace([1])
+  # a1                    # => [1]
+  # a1.remaining_capacity # => 3
+  # a1                    # => [1]
+  # a2.replace([1, 2, 3])
```
Comment (oprypin) [inline src/array.cr:1393]: What is `a2`? The overall code sample would fail not knowing what this variable is

Comment (wontruefree) [reply src/array.cr:1394]: solve https://github.com/crystal-lang/crystal/pull/11682/commits/703f6f4930e1d48671f099da81a4a953fbb26515

File: src/array.cr [hunk 4]
```diff
@@ -1381,6 +1381,19 @@ class Array(T)
...
 
+  # Replaces the contents of `self` with the contents of *other*.
+  # This resizes the Array to a greater capacity but does not free memory if the given array is smaller.
+  #
+  # ```
+  # a1 = [1, 2, 3]
+  # a1.replace([1])
+  # a1                    # => [1]
+  # a1.remaining_capacity # => 3
+  # a1                    # => [1]
```
Comment (beta-ziliani) [inline src/array.cr:1392]: This is redundant (and its obvious that `remaining_capacity` doesn't alter the array)

```suggestion
```

**Review by beta-ziliani — COMMENTED** [review]
Thanks @wontruefree , I left a minor comment

**Review by oprypin — APPROVED** [review]

**Review by straight-shoota — APPROVED** [review]

**Pull request merged**